pkg/controllers/customeradmin/group_controller.go

```diff
@@ -0,0 +1,132 @@
+package customeradmin
+
+import (
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	userv1client "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	osaCustomerAdmins  = "osa-customer-admins"
+	osaCustomerReaders = "osa-customer-readers"
+)
+
+type reconcileGroup struct {
+	osclient  *openshiftClient
+	aadClient *graphrbac.GroupsClient
+	aadConfig map[string]string
+	log       *logrus.Entry
+	groupMap  map[string]string
+	Admins    string `yaml:"admins,omitempty"`
+	Readers   string `yaml:"readers,omitempty"`
+}
+
+var _ reconcile.Reconciler = &reconcileGroup{}
+
+type openshiftClient struct {
+	CoreV1 corev1client.CoreV1Interface
+	UserV1 userv1client.UserV1Interface
+}
+
+func addGroupController(log *logrus.Entry, m manager.Manager) error {
+	r := &reconcileGroup{
+		osclient: &openshiftClient{
+			UserV1: userv1client.NewForConfigOrDie(m.GetConfig()),
+			CoreV1: corev1client.NewForConfigOrDie(m.GetConfig())},
+		log:      log,
+		groupMap: map[string]string{},
+	}
+	err := r.init()
+	if err != nil {
+		return err
+	}
+
+	r.aadClient, err = newAADGroupsClient(r.aadConfig)
+	if err != nil {
+		return err
+	}
+
+	c, err := controller.New("customeradmin-group-controller", m, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	if len(r.groupMap) == 0 {
+		log.Info("No group mapping config to sync")
+		return nil
+	}
+	events := make(chan event.GenericEvent)
+	timerSource := source.Channel{Source: events}
+	ticker := time.NewTicker(60 * time.Second)
+	quit := make(chan struct{})
+	timerSource.InjectStopChannel(quit)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				events <- event.GenericEvent{}
+			case <-quit:
+				log.Info("shutting down ticker")
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+	return c.Watch(&timerSource, &handler.EnqueueRequestForObject{}, &predicate.Funcs{GenericFunc: r.pollEvent})
+}
+
+func (r *reconcileGroup) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// not actually used (pollEvent is the real callback) but the controller.New() really wants it.
+	return reconcile.Result{}, nil
+}
+
+func (r *reconcileGroup) init() error {
+	aadGroupMap, err := r.osclient.CoreV1.ConfigMaps("kube-service-catalog").Get("aad-group-mapping", meta_v1.GetOptions{})
```

Comment:
You shouldn't use the kubernetes client to do this.  Expect your config file to be placed on disk for you and read it - this facilitates standalone testing.
The kubernetes deployment which runs this should have a volume configured which will lay down the Secret in the right place in the filesystem for you.
Rather than having a configmap and a secret (and somewhat overloading the cloudprovider config struct), better to define a brand new config format that this component will use.  Crib all of this from metricsbridge.